Add Retry Policy section to Jargon Mesh (#493)
temporalio/documentation: docs/jargon/mesh.md

- Although a Local Activity consumes less resources than a normal [**Activity**](#activity), it is subject to shorter durations and a lack of rate limiting.

## Retry Policy

A collection of attributes that instructs Temporal Server how to retry a failure of an Activity Task Execution or a [**Workflow Execution**](#workflow-execution).

> Any custom Retry Policy must be provided when an [**Activity Task**](#activity-task) or Workflow starts.

Author: Good point. I'll ax the link for now. We'll need to retrofit links for a number of terms later anyway.

Retrying an [**Activity Task**](#activity-task) produces a new Activity Task Execution that uses the same parameters [VERIFY] and Task Queue as the previous attempt.

An Activity Task Execution has a default Retry Policy and thus retries by default.

> A [**Workflow Execution**](#workflow-execution) does not have a default Retry Policy and thus does not retry by default.

Reviewer: ```suggestion
[**Workflow Executions**](#workflow-execution) do not start with a default Retry Policy and thus does not retry by default.
```

The intention is that a Workflow is written to never fail due to intermittent issues; an Activity is designed to handle such issues.
(Be aware that, by default, a Workflow Task Execution does retry indefinitely.)

A [**Workflow Execution**](#workflow-execution) should be retried only in specific scenarios, such as the following:

- A Cron Workflow or some other stateless always-running Workflow Execution that can benefit from retries.
- A file-processing or media-encoding Workflow Execution that downloads files to a host.

A Retry Policy does not apply to a Workflow Task Execution, which is governed by the Schedule-To-Start timeout.

> The time between retries is the _retry interval_. A retry interval is the smaller of two values:

Reviewer: ```suggestion
The time between retries is the _retry interval_.
A retry interval is the smaller of two values:
```

Reviewer: Shall we include the simple diagram here?

Author: Makes sense to me. @dtornow?

- The [initial interval](#initial-interval) multiplied by the [backoff coefficient](#backoff-coefficient) raised to power of the number of retries.
- The [maximum interval](#maximum-interval).

> A Retry Policy contains the following settings:

Reviewer: ```suggestion
A Retry Policy contains the following attributes:
```

- Although a Local Activity consumes less resources than a normal [**Activity**](#activity), it is subject to shorter durations and a lack of rate limiting.

## Retry Policy

A collection of attributes that instructs the Temporal Server how to retry a failure of an [Activity Task Execution](#activity-task-execution) or a [Workflow Execution](#workflow-execution).

- If a custom Retry Policy is to be used, it must be provided as an options parameter when an [Activity Execution](#activity-execution) or [Workflow Execution](#workflow-execution) is invoked.

> - The time between a retry is the _retry interval_.

Author: I previously had "The time between retries...", but that leaves out the first retry interval, which is between the initial attempt and the first retry attempt. My next commit changes the phrase to "The wait time before a retry...."

  A retry interval is the smaller of two values:
  - The [Initial Interval](#initial-interval) multiplied by the [Backoff Coefficient](#backoff-coefficient) raised to the power of the number of retries.
  - The [Maximum Interval](#maximum-interval).

> ![retry interval](/img/retry-interval-diagram.png)

Author: We need to add Maximum Interval to the formula in this diagram, @dtornow .
